sql/src/main/java/org/apache/druid/sql/calcite/external/ExternalOperatorConversion.java

```diff
@@ -106,28 +106,72 @@ public ExternalTableSpec apply(
         }
         final RowSignature rowSignature;
         if (columns != null) {
-          rowSignature = Columns.convertSignature(columns);
+          try {
+            rowSignature = Columns.convertSignature(columns);
+          }
+          catch (IAE e) {
+            throw new ArgumentAndException("columns", e);
+          }
```

Comment:
Instead of this throw/catch to just throw/catch again, why not create a private method

```
public static InvalidInput badArgumentException(Throwable cause, String fieldName)
{
  return InvalidInput.exception(
            cause,
            "Invalid value for the field [%s]. Reason: [%s]",
            fieldName,
            e.getMessage()
        );
}
```

Or, perhaps better yet would be to make the place that's throwing the IAE actually throw the right DruidException instead.  You can do a `try/catch (DruidException e) { throw e.prependAndBuild("Problem converting columns") }`

Reply:
Seems a lot cleaner. Since the places throwing exceptions are primarily object mappers, I added the static method for creating the appropriate druid exception message.